src/sage/matrix/matrix_modn_dense_template.pxi

```diff
@@ -441,14 +441,18 @@ cpdef __matrix_from_rows_of_matrices(X):
 
 
 cdef class Matrix_modn_dense_template(Matrix_dense):
-    def __cinit__(self):
+    def __cinit__(self, *args, bint zeroed_alloc=True, **kwds):
         cdef long p = self._base_ring.characteristic()
         self.p = p
         if p >= MAX_MODULUS:
             raise OverflowError("p (=%s) must be < %s."%(p, MAX_MODULUS))
 
-        self._entries = <celement *>check_allocarray(self._nrows * self._ncols, sizeof(celement))
-        self._matrix = <celement **>check_allocarray(self._nrows, sizeof(celement*))
+        if zeroed_alloc:
+            self._entries = <celement *>check_calloc(self._nrows * self._ncols, sizeof(celement))
+            self._matrix = <celement **>check_calloc(self._nrows, sizeof(celement*))
```

Comment:
`_matrix` never needs `calloc` since it is filled with values just after this allocation